### PR TITLE
New Possible Battle Outcomes

### PR DESCRIPTION
**Pull request by pepper-boi**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -158,24 +158,55 @@ async def bot_info(interaction: discord.Interaction):
...
+    
     async with interaction.channel.typing():
         await asyncio.sleep(1)
+        
+        
+    curr_season = server_config['season']
+
+    #Get leaderboard info
+    response = await rocketbot_client.query_leaderboard(curr_season)
+    response = json.loads(response['payload'])['records']
```
Comment (brianwinterpixel) [inline main.py:170]: if adding a query, the sleep code above could just be removed.

File: main.py [hunk 2]
```diff
@@ -158,24 +158,55 @@ async def bot_info(interaction: discord.Interaction):
...
+    events = {
+        "The bot dodged your attack. <:bot:917467970182189056>"
+        : 70,
+        "You destroyed the bot! It drops a single coin. <:coin:910247623787700264>"
+        : 10,
+        "The bot *expertly* dodged your attack. <:bot:917467970182189056>"
+        : 5,
+        "You thought you hit the bot, but its health returns to full due to network lag. 📶"
+        : 5,
+        "You destroyed the bot! It drops a some coins and a crate. <:coin:910247623787700264> <:coin:910247623787700264> <:coin:910247623787700264> 📦. But " + rand_player + " comes out of nowhere and steals it."
```
Comment (blakexe) [inline main.py:183]: If this is the only instance we use a random player we shouldn't be querying the leaderboard every time

File: main.py [hunk 3]
```diff
@@ -158,24 +158,55 @@ async def bot_info(interaction: discord.Interaction):
...
+        : 0.1,
+        "You get a quad kill, four birds one stone! It was four bots doing the same exact movement. They drop 4 coins. <:coin:910247623787700264> <:coin:910247623787700264> <:coin:910247623787700264> <:coin:910247623787700264>"
+        : 0.1,
+        "🗿 Moyai God comes down from the heavens and blocks your missile. You bow down (as a tank) and repent for your sins."
+        : 0.1,
+        }
+    event = "You fire a missile at a bot. <:rocketmint:910253491019202661>\n" + random.choices(population=events.keys(), weights=events.values(), k=1)[0]
```
Comment (blakexe) [inline main.py]: After the event has been randomized, you could do something like `event.startswith('You destroyed the bot! It drops a some coins and a crate.')` or replace that with a more unique start string to let you know you should be using the randomized player and then query the leaderboard and populate the string

**Review by blakexe — CHANGES_REQUESTED** [review]

File: main.py (hunk 3 above)
Comment (brianwinterpixel) [reply main.py]: see possible way to do this:
https://stackoverflow.com/a/16669571

**Pull request merged**